### ISSUE TITLE
canvas: Rick Dangerous doesn't display correctly

### ISSUE THREAD

**Issue by Toad06** [ruffle-rs/ruffle#7448]
- Disable WebGL in your browser (Firefox: set `webgl.disabled` to true ; Chrome: launch with `-disable-webgl`).
- Browse to http://rickdangerousflash.free.fr/
- Notice that there's something wrong. :)

Result:
![r1](https://user-images.githubusercontent.com/14274782/179796365-e2df2448-b6f5-4b0e-90cb-07770f26ec07.png)

Expected:
![r2](https://user-images.githubusercontent.com/14274782/179796371-43e07d07-d47f-4154-9fe6-39d97d034a03.png)

---
Unlike Chrome, this didn't use to happen in Firefox some months ago.
a10c2ff9c7878a936b22298ede41edf5eaec1338 seems to cause the regression. Reverting and extending this commit to Chrome prevents the issue from happening in both browsers.

**Comment by Toad06** [comment]
This issue also affects [Steppenwolf 1-2](https://www.y8.com/games/steppenwolf_chapter_1_episode_2) in a less critical way.

Current build:
![c1](https://user-images.githubusercontent.com/14274782/179800512-3072cb37-2598-4801-a45b-ca4d93b2f880.png)

Expected:
![c2](https://user-images.githubusercontent.com/14274782/179800516-936cb8fd-30f2-4afe-a46b-2154eafb2cf2.png)

**Comment by Herschel** [comment]
This is caused by additive alpha in a colortransform. Using a positive alpha component in an SVG `feColorMatrix` when drawing to a canvas ends up darkening the entire canvas, not just the drawn path. For example, a color transform with +128 alpha would draw 50% black on the entire canvas. This seems to be a bug, or at least surprising behavior. I filed a bug on Chromium here: https://bugs.chromium.org/p/chromium/issues/detail?id=1327508

A workaround is possible by tacking on an `<feComponentTransfer>` filter that zeroes out any alpha below the threshold: https://jsfiddle.net/mwelsh/qmL9x0up/35/

**Comment by Toad06** [comment]
Are you sure that this is completely the same issue? I'm asking because these games also have the same rendering issue in Firefox since a10c2ff9c7878a936b22298ede41edf5eaec1338 (sorry if I didn't make this point clear enough in the first post) but the bug report you made on Chromium shows that Firefox shouldn't be affected. Sorry if I misunderstood something.

**Comment by Lord-McSweeney** [comment]
Unfortunately still happening.

**Comment by Toad06** [comment]
Yes, this issue could be fixed by re-using the hack removed in a10c2ff9c7878a936b22298ede41edf5eaec1338 (for Firefox... and also Chrome) but I couldn't explain why.

**Comment by artem21072015** [comment]
> This issue also affects [Steppenwolf 1-2](https://www.y8.com/games/steppenwolf_chapter_1_episode_2) in a less critical way.
> 
> Current build: ![c1](https://user-images.githubusercontent.com/14274782/179800512-3072cb37-2598-4801-a45b-ca4d93b2f880.png)
> 
> Expected: ![c2](https://user-images.githubusercontent.com/14274782/179800516-936cb8fd-30f2-4afe-a46b-2154eafb2cf2.png)

ngl the broken one looks better than expected
